src/samples/audio/detect_test.c

```diff
@@ -140,6 +151,62 @@ static void notify_host(const struct comp_dev *dev)
 #endif /* CONFIG_IPC_MAJOR_4 */
 }
 
+#if CONFIG_AMS
+static int test_keyword_register_ams_producer(const struct comp_dev *dev)
+{
+	struct comp_data *cd = comp_get_drvdata(dev);
+	const uint8_t *ams_kpd_msg = (const uint8_t *)SOF_RT_UUID(ams_kpd_msg_uuid);
+	uint16_t mod_id, inst_id;
+	int ret;
+
+	comp_dbg(dev, "test_keyword_register_ams_producer()");
+
+	mod_id = IPC4_MOD_ID(dev_comp_id(dev));
+	inst_id = IPC4_INST_ID(dev_comp_id(dev));
+
+	ret = ams_get_message_type_id(ams_kpd_msg, &cd->ams_uuid_id);
+	if (ret)
+		return ret;
+
+	ret = ams_register_producer(cd->ams_uuid_id, mod_id, inst_id);
+	if (ret)
+		return ret;
+
+	cd->ams_payload.message_type_id = cd->ams_uuid_id;
+	cd->ams_payload.producer_module_id = mod_id;
+	cd->ams_payload.producer_instance_id = inst_id;
+	cd->ams_payload.message_length = sizeof(struct kpb_client);
+	cd->ams_payload.message = (uint8_t *)&cd->client_data;
```

Comment:
would it make sense to make an exported `kpb_register_ams_producer(const struct comp_dev *dev, struct ams_message_payload *payload)` and only let the caller add `payload->message` and `payload->message_length` after calling it?